### PR TITLE
Team and Group pages pass 1

### PR DESCRIPTION
**Pull request by srkirkland**
Created team list, team detail, and folder detail pages.  Includes some ugly react pages and decent API methods to feed all the necessary info.  No membership management or CRUD functionality.

Created new `Folder` and `Team` controllers to handle API.  Added some helpers to simplify getting user Iam and use that for all permissions checks.  In react, added responseModel types to `types.ts` -- perhaps we might want to separate out later.  Also created a few actual folders to organize a little better pages and components so it doesn't get to be one large pages folder.

### CODE REVIEW

File: Finjector.Web/ClientApp/src/pages/Teams/Team.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        setSearch={setSearch}
+      />
+
+      <h3>Team name: {teamModel.data?.team.name}</h3>
+      <FolderList teamModel={teamModel.data} filter={search} />
+      <button
+        className="btn btn-outline-secondary flex-fill me-3"
+        type="button"
+      >
+        Create New Folder
```
Comment (jSylvestre) [inline Finjector.Web/ClientApp/src/pages/Teams/Team.tsx:29]: Create new Group

File: Finjector.Web/ClientApp/src/components/Teams/FolderList.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  return (
+    <ul className="list-group">
+      {filteredFolderInfo.map((folderInfo) => (
+        <li
+          className="fin-item d-flex justify-content-between align-items-center saved-list-item"
+          key={folderInfo.folder.id}
+        >
+          <div className="col-9 ms-2 me-auto">
+            <div className="fw-bold "> {folderInfo.folder.name}</div>
+            <Link to={`/teams/${teamModel.team.id}/folders/${folderInfo.folder.id}`}>Go to Folder</Link>
```
Comment (jSylvestre) [inline Finjector.Web/ClientApp/src/components/Teams/FolderList.tsx:35]: Go to Group

File: Finjector.Web/ClientApp/src/pages/Teams/Folder.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+  return (
+    <div>
+      <SearchBar
+        placeholderText="Search Within Group"
+        search={search}
+        setSearch={setSearch}
+      />
+
+      <h3>Folder name: {folderModel.data?.folder.name}</h3>
```
Comment (jSylvestre) [inline Finjector.Web/ClientApp/src/pages/Teams/Folder.tsx:23]: Group Name

**Review by jSylvestre — CHANGES_REQUESTED** [review]

File: Finjector.Web/ClientApp/src/pages/Teams/Team.tsx (hunk 1 above)
Comment (srkirkland) [reply Finjector.Web/ClientApp/src/pages/Teams/Team.tsx:29]: @jSylvestre I agree with all these changes in principle but this is really just for Cal to take over and redo to his desires.  So I'm just focusing on getting the data and don't really care about any buttons or text or anything

**Review by jSylvestre — APPROVED** [review]

**Pull request merged**